suite-native/device-manager/src/components/DeviceList.tsx

```diff
@@ -0,0 +1,165 @@
+
+import {
+    ConnectDeviceStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+import { getSortedDevicesWithoutInstances } from '@suite-common/suite-utils';
+import { analytics, EventType } from '@suite-native/analytics';
+import {
+    DeviceRootState,
+    PORTFOLIO_TRACKER_DEVICE_ID,
+    selectDevice,
+    selectDeviceById,
+    selectPhysicalDevices,
+} from '@suite-common/wallet-core';
+import { Button, HStack, VStack, Text, Divider, Box } from '@suite-native/atoms';
+import { AcquiredDevice, TrezorDevice } from '@suite-common/suite-types';
+import { Translation } from '@suite-native/intl';
+import { prepareNativeStyle, useNativeStyles } from '@trezor/styles';
+
+import { DeviceItem } from './DeviceItem';
+import { useDeviceManager } from '../hooks/useDeviceManager';
+import { MANAGER_MODAL_BOTTOM_RADIUS } from './DeviceManagerModal';
+
+type NavigationProp = StackToStackCompositeNavigationProps<
+    RootStackParamList,
+    RootStackRoutes.AppTabs,
+    RootStackParamList
+>;
+
+type DeviceListProps = {
+    isVisible: boolean;
+    onSelectedDevice?: (device: AcquiredDevice) => void;
+};
+
+const ITEM_HEIGHT = 62;
+const SPACING = 8;
+const BUTTON_HEIGHT = 48;
+const SEPARATOR_HEIGHT = 18;
+const VERTICAL_PADDING = 16;
+
+const buttonWrapperStyle = prepareNativeStyle(utils => ({
+    paddingHorizontal: utils.spacings.medium,
+}));
+
+export const DeviceList = ({ isVisible, onSelectedDevice }: DeviceListProps) => {
+    const { applyStyle } = useNativeStyles();
+    const navigation = useNavigation<NavigationProp>();
+    const { setIsDeviceManagerVisible } = useDeviceManager();
+    const device = useSelector(selectDevice);
+    const physicalDevices = useSelector(selectPhysicalDevices);
+    const portfolioDevice = useSelector((state: DeviceRootState) =>
+        selectDeviceById(state, PORTFOLIO_TRACKER_DEVICE_ID),
+    );
+    const { utils } = useNativeStyles();
+    const opacity = useSharedValue(0);
+    const height = useSharedValue(0);
+    const [isShown, setIsShown] = useState(false);
+
+    const allDevices = [...physicalDevices, portfolioDevice].filter(
+        a => a !== undefined,
+    ) as TrezorDevice[];
+    const otherDevices = getSortedDevicesWithoutInstances(allDevices, device?.id);
+
+    const handleConnectDevice = () => {
+        if (device) {
+            onSelectedDevice?.(device as AcquiredDevice);
+        }
+        setIsDeviceManagerVisible(false);
+        navigation.navigate(RootStackRoutes.ConnectDeviceStack, {
+            screen: ConnectDeviceStackRoutes.ConnectAndUnlockDevice,
+        });
+        analytics.report({
+            type: EventType.DeviceManagerClick,
+            payload: { action: 'connectDeviceButton' },
+        });
+    };
+
+    //to hide/show the list with animation
+    useEffect(() => {
+        let timer: ReturnType<typeof setTimeout>;
+
+        if (isVisible) {
+            const h =
```

Comment:
this is very long expression. Could you split it into multiple for better readability please?

Reply:
5825a46c34